x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/field_select.tsx

```diff
@@ -48,25 +48,41 @@ export function FieldSelect({
   );
 
   function isCompatibleWithCurrentOperation(col: BaseIndexPatternColumn) {
-    return incompatibleSelectedOperationType
-      ? col.operationType === incompatibleSelectedOperationType
-      : !selectedColumn || col.operationType === selectedColumn.operationType;
+    if (incompatibleSelectedOperationType) {
+      return col.operationType === incompatibleSelectedOperationType;
+    }
+    return !selectedColumn || col.operationType === selectedColumn.operationType;
   }
 
   const fieldOptions = [];
-  const fieldLessColumn = filteredColumns.find(column => !hasField(column));
-  if (fieldLessColumn) {
+  const bestFieldLessColumn = filteredColumns.find(
```

Comment:
Nit: FieldLess should probably be Fieldless (one word), since I think this is trying to convey "columns which have no associated field". Field less column sounds to me like a mathematical subtraction: "subtract column from field"

Reply:
Agreed